fennel/datasets/datasets.py

```diff
@@ -473,6 +475,13 @@ def first(self) -> _Node:
             self.keys = self.keys[0]  # type: ignore
         return First(self.node, list(self.keys))  # type: ignore
 
+    def window(self, type: str, gap: str, field: str) -> _Node:
+        if len(self.keys) == 1 and isinstance(self.keys[0], list):
+            self.keys = self.keys[0]  # type: ignore
+        return WindowOperator(
```

Comment:
Can you also add a check that field should not be among the list of columns already present ?

Reply:
Yup, will add this.